test/cpp/api/serialize.cpp

```diff
@@ -667,6 +667,107 @@ TEST(SerializeTest, Optim_RMSprop) {
   is_optimizer_state_equal<RMSpropParamState>(optim1.state(), optim1_2.state());
 }
 
+TEST(SerializeTest, Optim_LBFGS) {
+  auto options = LBFGSOptions();
+  using OptimizerClass = LBFGS;
+  using DerivedOptimizerOptions = LBFGSOptions;
+  using DerivedOptimizerParamState = LBFGSParamState;
+  auto model1 = Linear(5, 2);
+  auto model2 = Linear(5, 2);
+  auto model3 = Linear(5, 2);
+
+  // Models 1, 2, 3 will have the same parameters.
+  auto model_tempfile = c10::make_tempfile();
+  torch::save(model1, model_tempfile.name);
+  torch::load(model2, model_tempfile.name);
+  torch::load(model3, model_tempfile.name);
+
+  auto param1 = model1->named_parameters();
+  auto param2 = model2->named_parameters();
+  auto param3 = model3->named_parameters();
+  for (const auto& p : param1) {
+    ASSERT_TRUE(p->allclose(param2[p.key()]));
+    ASSERT_TRUE(param2[p.key()].allclose(param3[p.key()]));
+  }
+  // Make some optimizers
+  auto optim1 = OptimizerClass(
+      {torch::optim::OptimizerParamGroup(model1->parameters())}, options);
+  auto optim2 = OptimizerClass(
+      model2->parameters(), options);
+  auto optim2_2 = OptimizerClass(
+      model2->parameters(), options);
+  auto optim3 = OptimizerClass(
+      model3->parameters(), options);
+  auto optim3_2 = OptimizerClass(
+      model3->parameters(), options);
+
+  auto x = torch::ones({10, 5});
+
+  auto step = [&x](torch::optim::LossClosureOptimizer& optimizer, Linear model) {
+    optimizer.zero_grad();
+    auto y = model->forward(x).sum();
```

Comment:
I think we would need `y.backward()` after this line to propagate the gradients to the parameters of `model`

Reply:
yes :D